.github/workflows/rust.yml

```diff
@@ -184,11 +184,10 @@ jobs:
   autodeploy:
     runs-on: ubuntu-latest
     needs: [deploy]
-    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
     steps:
     - uses: actions/checkout@v2
     - name: Deploy staging
-      if: github.ref == 'refs/heads/main'
+      if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
I would move this condition to right after `needs` since we don't need to checkout the branch or start this job if the branch is not `main`